src/preprocess.py

```diff
@@ -178,6 +179,47 @@ def construct_documents_mapping(collection: OrdDict[str, Any]) -> OrdDict[int, s
         mapping[idx] = key
     return mapping
 
+@dataclass
+class StatCollection:
+    """
+    Usefull stats of a collection:
+
+    'self.nb_docs' is the number of docs in the collection
+    'self.doc_stats' is a dictionary that provides for a given article id its collection
```

Comment:
'self.doc_stats' is a dictionary that provides for a given article id its *collection* ?

![image](https://user-images.githubusercontent.com/1516762/79465542-c2351900-7ffb-11ea-8709-306e1d1196f4.png)

Reply:
done